backend/src/layer/data/mail/mail_sender.rs

```diff
@@ -0,0 +1,171 @@
+#[derive(Debug, Error)]
+pub enum MailError {
+    #[error("an error occurred while parsing the addresses: {0}")]
+    AddressError(#[from] AddressError),
+    #[error("an error occurred while reading the template: {0}")]
+    TemplateError(#[from] std::io::Error),
+    #[error("an error occurred while parsing the mail: {0}")]
+    MailParseError(#[from] lettre::error::Error),
+    #[error("an error occurred while sending the mail: {0}")]
+    MailSendError(#[from] lettre::transport::smtp::Error),
+}
+
+pub struct MailSender {
+    config: MailInfo,
+    mailer: SmtpTransport,
+}
+
+#[async_trait]
+impl AdminNotification for MailSender {
+    async fn notify_admin_image_report(&self, info: ImageReportInfo) {
+        if let Err(error) = self.try_notify_admin_image_report(&info) {
+            warn!("{error:?}");
+        }
+    }
+}
+
+impl MailSender {
+    /// Creates a new [`MailSender`] with the attributes defined in config. Also creates an SMTP connection to the smtp server defined in config
+    ///
+    /// # Errors
+    /// Returns an error, if the connection could not be established to the smtp server
+    pub fn new(config: MailInfo) -> Result<Self> {
+        let creds = Credentials::new(config.username.clone(), config.password.clone());
+        if let Ok(transport_builder) = SmtpTransport::relay(&config.smtp_server) {
+            let mailer = transport_builder
+                .port(config.smtp_port)
+                .credentials(creds)
+                .build();
+            Ok(Self { config, mailer })
+        } else {
+            Err(DataError::NoSuchItem)
+        }
+    }
+
+    fn try_notify_admin_image_report(&self, info: &ImageReportInfo) -> MailResult<()> {
+        let sender = self.get_sender()?;
+        let reciever = self.get_reciever()?;
+        let report = Self::get_report(info)?;
+        let email = Message::builder()
+            .from(sender)
+            .to(reciever)
+            .subject("An image was reported for reviewing")
+            .body(report)?;
+        self.mailer.send(&email)?;
+        info!("Email sent successfully!");
+        Ok(())
+    }
+
+    fn get_sender(&self) -> MailResult<Mailbox> {
+        format!("app <{}>", self.config.username.clone())
+            .parse()
+            .map_err(MailError::AddressError)
+    }
+
+    fn get_reciever(&self) -> MailResult<Mailbox> {
+        format!("admin <{}>", self.config.admin_email_address)
+            .parse()
+            .map_err(MailError::AddressError)
+    }
+
+    fn get_report(info: &ImageReportInfo) -> MailResult<String> {
+        let template_file_contents =
+            fs::read_to_string(REPORT_TEMPLATE_FILE).map_err(MailError::TemplateError)?;
+        let template = Template::new(&template_file_contents);
+        let mut args = HashMap::new();
+        let image_link: &str = &info.image_link;
+        args.insert("image_link", image_link);
+        let image_id: &str = &Simple::from_uuid(info.image_id).to_string();
```

Comment:
why not just `.to_string()` using hyphonated formatting?
https://docs.rs/uuid/latest/uuid/struct.Uuid.html#formatting